Ska/engarchive/fetch.py

```diff
@@ -2189,3 +2192,117 @@ def _plural(x):
     known small set of cases within fetch where it will get applied.
     """
     return x + "es" if (x.endswith("x") or x.endswith("s")) else x + "s"
+
+
+def get_data_gap_spec_parser():
+    """
+    Get parser for fetch data gap specification.
+
+    This env var is in the form of a command line argument string with the following
+    command line options::
+
+        --include INCLUDE  Include MSIDs matching glob (default="*", can be repeated)
+        --exclude EXCLUDE  Exclude MSIDs matching glob (default=None, can be repeated)
+        --start START      Gap start (relative seconds or abs time)
+        --stop STOP        Gap stop (relative seconds or abs time)
+
+    For example::
+
+        "--exclude=*EPHEM* --start=-25000 --stop=-2000"
+
+    """
+    import argparse
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--include",
+        default=[],
+        action="append",
+        help="Include MSIDs matching glob (default='*', can be repeated)",
+    )
+    parser.add_argument(
+        "--exclude",
+        default=[],
+        action="append",
+        help="Exclude MSIDs matching glob (default=None, can be repeated)",
+    )
+    parser.add_argument(
+        "--start", default=-30000, help="Gap start (relative seconds or abs time)"
+    )
+    parser.add_argument(
+        "--stop", default=-3000, help="Gap stop (relative seconds or abs time)"
+    )
+    return parser
+
+
+def msid_matches_data_gap_spec(msid, includes, excludes):
+    from fnmatch import fnmatch
+
+    msid = msid.upper()
+    includes = includes or ["*"]
+    match = any(fnmatch(msid, include.upper()) for include in includes) and not any(
+        fnmatch(msid, exclude.upper()) for exclude in excludes
+    )
+    return match
+
+
+def create_msid_data_gap(msid_obj: MSID, data_gap_spec: str):
+    """
+    Make a data gap in the ``msid_obj`` (in-place) by removing data points.
+
+    This is mostly useful for testing via setting the ``CHETA_FETCH_DATA_GAP``
```

Comment:
Are there other applications besides testing?  I was wondering if this belongs in tests.

Reply:
For kadi this is useful for functional testing of the validation application, so that implies effectively being part of the public API for cheta.